### PR TITLE
Conversion from u8 to stdint uint8_t style types

### PR DESCRIPTION
**Pull request by abrauchli**
Using the Linux style u8, s16, etc. types has historically been an issue when compiling against certain arduino configurations. As such, we're moving away from those to the stdint types.

### CODE REVIEW

**Comment by abrauchli** [comment]
There's no CI on embedded-common, but I test compiled locally with embedded-scd

**Comment by rnestler** [comment]
> There's no CI on embedded-common, but I test compiled locally with embedded-scd

Should we add it? A basic compile test should be easily possible.

File: README.md [hunk 1]
```diff
@@ -4,8 +4,10 @@ code. It is a dependency of the Sensirion embedded drivers and it should not be
 necessary to clone it independently.
 
 We recommend to download the repository recursively within the
-[SGP](https://github.com/Sensirion/embedded-sgp) and
-[SHT](https://github.com/Sensirion/embedded-sht) projects.
+[SCD](https://github.com/Sensirion/embedded-scd),
+[SGP](https://github.com/Sensirion/embedded-sgp),
+[SGP](https://github.com/Sensirion/embedded-sht) or
+[SHT](https://github.com/Sensirion/embedded-sps) projects.
```
Comment (rnestler) [inline README.md]: Link names do not match the targets.

Comment (abrauchli) [reply README.md]: done

**Comment by abrauchli** [comment]
> > There's no CI on embedded-common, but I test compiled locally with embedded-scd
> 
> Should we add it? A basic compile test should be easily possible.

It doesn't belong in this PR, and it would solve itself by merging.. another reason for unifying.

**Comment by abrauchli** [comment]
(last change is rebase on master, no changes)

**Pull request merged**